programs/budget_api/src/budget_transaction.rs

```diff
@@ -69,11 +72,13 @@ impl BudgetTransaction {
         recent_blockhash: Hash,
     ) -> Transaction {
         let from = from_keypair.pubkey();
-        TransactionBuilder::default()
+        let mut tx = TransactionBuilder::default()
             .push(BudgetInstruction::new_apply_timestamp(
                 &from, contract, to, dt,
             ))
-            .sign(&[from_keypair], recent_blockhash)
+            .compile();
+        tx.sign(&[from_keypair], recent_blockhash);
```

Comment:
wdyt about `.sign()` returning the Transaction?

```rust
    fn foo(...) -> Transaction {
         TransactionBuilder::default()
            .push(BudgetInstruction::new_apply_timestamp(
                &from, contract, to, dt,
            ))
            .compile()
            .sign(&[from_keypair], recent_blockhash)
    }
```

Reply:
It won't make sense to do that once we have ThinClient signing pre-existing transactions.

Reply:
And more generally, doing a mutable operation and returning the thing you mutated isn't too cool by functional programming standards.  I assume that's why `Vec::sort()` doesn't return itself, for example.